registry/api/v2/names_test.go

```diff
@@ -6,14 +6,23 @@ import (
 	"testing"
 )
 
-func TestRepositoryNameRegexp(t *testing.T) {
-	for _, testcase := range []struct {
+var (
+	// regexpTestcases is a unified set of testcases for
+	// TestValidateRepositoryName and TestRepositoryNameRegexp.
+	// Some of them are valid inputs for one and not the other.
+	regexpTestcases = []struct {
+		// input is the repository name or name component testcase
 		input string
-		err   error
+		// errFromValidateRepositoryName is the error expected from
+		// ValidateRepositoryName, or nil
+		errFromValidateRepositoryName error
```

Comment:
Please see [Rule 6 in coding style](https://github.com/docker/distribution/blob/master/CONTRIBUTING.md#coding-style).